python/python/tests/test_async_client.py

```diff
@@ -5201,25 +5209,281 @@ async def test_xgroup_create_consumer_xgroup_del_consumer(
+                key, [("f1_3", "v1_3")], StreamAddOptions(stream_id1_3)
+            )
+            == stream_id1_3
+        )
+        # since NOACK is passed, stream entry will be consumed without being added to the pending entries
+        assert await redis_client.xreadgroup(
+            {key: ">"},
+            group_name,
+            consumer_name,
+            StreamReadGroupOptions(no_ack=True, count=5, block_ms=1000),
+        ) == {key: {stream_id1_3: [["f1_3", "v1_3"]]}}
+        assert (
+            await redis_client.xreadgroup(
+                {key: ">"},
+                group_name,
+                consumer_name,
+                StreamReadGroupOptions(no_ack=False, count=5, block_ms=1000),
+            )
+            is None
+        )
+        assert await redis_client.xreadgroup(
+            {key: "0"},
+            group_name,
+            consumer_name,
+            StreamReadGroupOptions(no_ack=False, count=5, block_ms=1000),
+        ) == {key: {}}
 
         # attempting to call XGROUP CREATECONSUMER or XGROUP DELCONSUMER with a non-existing key should raise an error
         with pytest.raises(RequestError):
             await redis_client.xgroup_create_consumer(
-                non_existing_key, group_name, consumer
+                non_existing_key, group_name, consumer_name
             )
         with pytest.raises(RequestError):
             await redis_client.xgroup_del_consumer(
-                non_existing_key, group_name, consumer
+                non_existing_key, group_name, consumer_name
             )
 
         # key exists, but it is not a stream
         assert await redis_client.set(string_key, "foo") == OK
         with pytest.raises(RequestError):
-            await redis_client.xgroup_create_consumer(string_key, group_name, consumer)
+            await redis_client.xgroup_create_consumer(
+                string_key, group_name, consumer_name
+            )
+        with pytest.raises(RequestError):
+            await redis_client.xgroup_del_consumer(
+                string_key, group_name, consumer_name
+            )
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_xreadgroup_edge_cases_and_failures(
+        self, redis_client: TRedisClient, cluster_mode, protocol, request
+    ):
+        key = f"{{testKey}}:{get_random_string(10)}"
+        non_existing_key = f"{{testKey}}:{get_random_string(10)}"
+        string_key = f"{{testKey}}:{get_random_string(10)}"
+        group_name = get_random_string(10)
+        consumer_name = get_random_string(10)
+        stream_id0 = "0"
+        stream_id1_0 = "1-0"
+        stream_id1_1 = "1-1"
+
+        # attempting to execute against a non-existing key results in an error
         with pytest.raises(RequestError):
-            await redis_client.xgroup_del_consumer(string_key, group_name, consumer)
```

Comment:
did you intend to delete the `xgroup_del_consumer` test case?

Reply:
It isn't actually deleted, Git is just getting mixed up identifying the diffs. On line 5319 above the xgroup_del_consumer is labelled as a new added line, but it should really be labelled as replacing the line thats "deleted" here

Reply:
Got it... I missed the added line.  Thanks.